docs/gcloud-auth.rst

```diff
@@ -0,0 +1,208 @@
+.. toctree::
+  :maxdepth: 1
+  :hidden:
+
+Authentication
+--------------
+
+=======
+Summary
+=======
+
+For the majority of cases,
+authentication should "just work"
+as credentials can be loaded implicitly
+based on the environment you're application is running in
+and therefore there is no extra code required to authenticate.
```

Comment:
I think we should add something like
```
Note that for certain environments there are additional configuration steps needed, refer to the section below that explains other needed steps for your environment.
```

Those steps are:
* For compute engine, you have to create the instance with appropriate scopes.
* Create service account, download json, etc

I would also add that, in any cases, the service account should have appropriate permissions on your API. (FYI, some old app engine apps don't have the app engine's service account in the permission list. Also, you can delete those default service accounts from the permission list manually.)